cpp/src/arrow/ipc/ipc-json-test.cc

```diff
@@ -0,0 +1,279 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
For the sake of documentation and understanding, it would be nice to have a test that will read a JSON from a string embedded in the code here.  Not sure how large the minimal possible version could be but that would be really helpful to understand the format.